src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -575,33 +575,14 @@ private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool
                             ScanInterpolatedStringLiteralNestedString();
                             continue;
                         case '@':
-                            if (_lexer.TextWindow.PeekChar(1) == '"' && !RecoveringFromRunawayLexing())
```

Comment:
>!RecoveringFromRunawayLexing() [](http://example.com/codeflow?start=72&length=30)

Is this condition necessary? It doesn't appear to be include in `TryScanAtStringToken()`.

Reply:
i don't think it's necessary, and i think it's actively harmful.  if we've hit some issue while lexing an interpolated string, but we hit a `@"` we should still treat that as the start of a string.  It is much more likely to be than, rather than the end of some string that ends with `"...@"`.  Thus, if we skip the `@` and then treat the `"` as the end of hte string, rather than a start, we'll actually make things worse.